backend/btrixcloud/auth.py

```diff
@@ -171,10 +173,52 @@ def get_bearer_response(user: User):
     async def login(
         credentials: OAuth2PasswordRequestForm = Depends(),
     ):
+        """Prevent brute force password attacks.
+
+        After 5 or more consecutive failed login attempts for the same user,
+        lock the user account and send an email to reset their password.
+        On successful login when user is not already locked, reset count to 0.
+        """
         user = await user_manager.authenticate(
             credentials.username, credentials.password
         )
 
+        attempted_user = await user_manager.get_by_email(credentials.username)
+        if attempted_user is None:
```

Comment:
This theoretically would allow someone to discover if an email is registered in the system or not - if they always get a 400, then its not a valid user, but if they get a 429, then it is.
Not sure of a good way around this, though.

Reply:
Yeah I'm not sure a good way around this either, will think on it.

Reply:
With the FailedLogin object, we could can now add these regardless of whether its a valid email or not! So using the same invalid email 5 times will get a 429, but won't send a reset if its not a valid user.

Reply:
Added a test to cover this :)